src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs

```diff
@@ -17,10 +17,10 @@ private partial class Worker
         {
             private readonly struct ResolvedReferences
             {
-                public ImmutableArray<ProjectReference> ProjectReferences { get; }
+                public ImmutableHashSet<ProjectReference> ProjectReferences { get; }
```

Comment:
Are we concerned about ordering being blown out here?

Reply:
@DustinCampbell I wasn't concerned; I don't think reference ordering has any semantic meaning, and also wasn't sure if MSBuild and everything else is maintaining ordering anyways. Looking at the Workspace code, we do maintain the ordering of ProjectReferences, but give you no way to specify the full ordering of the references that go into the Compilation: we always add metadata references before project references.

Reply:
Per @jaredpar he doesn't think ordering matters.

Reply:
I said I didn't think ordering matters. I didn't say we should make business decisions on my opinion 😉